### DISCUSSION TITLE
Profiling

### DISCUSSION THREAD

**Discussion by PasVegan** [writer/writer-framework#133] [Q&A] [answered]
Hi,
How can I profile my app ? When I launch my app in profiling mode with Pycharm only the "server" get profiled so i can only see weird function call and not a single one from my code.

![Result](https://github.com/streamsync-cloud/streamsync/assets/114608503/98b29996-df6f-40d1-8a8e-b0b981202a3c)

[eSW-Dashboard5.pstat.zip](https://github.com/streamsync-cloud/streamsync/files/12586264/eSW-Dashboard5.pstat.zip)

**Comment by PasVegan** [comment]
nobody :( ?

**Comment by ramedina86** [comment]
Hi, I've never used Pycharm, but I can confirm is that your app runs in a different process, so you'll need support for multiprocessing profiling (not sure if Pycharm and the version you're using support this). You may find it under the name "AppProcess" as that's the name the process is initialised with.

**Reply by PasVegan** [reply]
Hi,
I tried using cProfile and yappi, Pycharm can use both of them. Did you succeed in profiling a streamsync app ?

**Comment by FabienArcellier** [comment, marked as answer]
I have shared a solution using `py-spy` on [this issue](https://github.com/streamsync-cloud/streamsync/issues/137#issuecomment-1732344426) and we discuss about [how to use ``inline profiling`` on the discord](https://discord.com/channels/1154359809454379088/1154359809454379091/1155174381379666081) on discord
